Nos/Service/EventProcessor.swift

```diff
@@ -23,14 +24,23 @@ enum EventProcessor {
     static func parse(
         jsonEvent: JSONEvent,
         from relay: Relay?,
-        in parseContext: NSManagedObjectContext
+        in parseContext: NSManagedObjectContext,
+        skipVerification: Bool = false
     ) throws -> Event {
         let event = try Event.findOrCreate(jsonEvent: jsonEvent, relay: relay, context: parseContext)
         
-        guard event.author?.publicKey != nil else {
+        guard let publicKey = event.author?.publicKey else {
             throw EventError.missingAuthor
         }
         
+        if skipVerification == false {
+            guard try publicKey.verifySignature(on: event) else {
+                parseContext.delete(event)
+                Log.info("Invalid signature on event: \(jsonEvent)")
+                throw EventError.invalidSignature(event)
+            }
```

Comment:
maybe we need to set `event.isVerified = true` here (outside the guard)

Reply:
Yeah good call. We aren't reading `isVerified` anywhere anymore but it's probably still good to keep track of for now. Maybe we can remove it in the future if we don't end up using it.